### PR TITLE
Update to latest version of spring

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -34,71 +19,45 @@ dependencies {
...
-  testImplementation("io.mockk:mockk:1.13.9")
-  testImplementation("org.mockito.kotlin:mockito-kotlin:5.2.0")
-  testImplementation("org.junit.jupiter:junit-jupiter")
+  testImplementation("io.mockk:mockk:1.13.11")
+  testImplementation("org.mockito.kotlin:mockito-kotlin:5.3.1")
   testImplementation("org.testcontainers:localstack")
-
-  testImplementation("io.kotest:kotest-runner-junit5:5.8.0")
-  testImplementation("io.kotest:kotest-assertions-core:5.8.0")
-  testImplementation("io.kotest:kotest-property:5.8.0")
```
Comment (andrewrlee) [inline build.gradle.kts:74]: not used

File: build.gradle.kts [hunk 2]
```diff
@@ -34,71 +19,45 @@ dependencies {
...
+  implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.5.0")
 
   implementation("org.apache.commons:commons-lang3:3.14.0")
-  implementation("commons-io:commons-io:2.15.1")
-  implementation("com.google.guava:guava:32.1.3-jre")
-
   implementation("org.ehcache:ehcache:3.10.8")
   implementation("com.zaxxer:HikariCP:5.1.0")
-  implementation("com.oracle.database.jdbc:ojdbc10:19.21.0.0")
-  implementation("org.mockito.kotlin:mockito-kotlin:5.2.1")
```
Comment (andrewrlee) [inline build.gradle.kts:57]: removed these as not used as non-test dependencies

File: build.gradle.kts [hunk 3]
```diff
@@ -34,71 +19,45 @@ dependencies {
...
 
-  implementation("commons-codec:commons-codec:1.16.0")
-  implementation("io.swagger:swagger-annotations:1.6.12")
-  implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.3.0")
+  implementation("commons-codec:commons-codec:1.17.0")
+  implementation("org.springdoc:springdoc-openapi-starter-webmvc-ui:2.5.0")
 
   implementation("org.apache.commons:commons-lang3:3.14.0")
-  implementation("commons-io:commons-io:2.15.1")
-  implementation("com.google.guava:guava:32.1.3-jre")
```
Comment (andrewrlee) [inline build.gradle.kts:52]: not used... I think this could be trimmed back further

File: build.gradle.kts [hunk 4]
```diff
@@ -1,28 +1,13 @@
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_21
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
-  id("uk.gov.justice.hmpps.gradle-spring-boot") version "5.10.1"
-  kotlin("plugin.spring") version "1.9.21"
-  kotlin("plugin.jpa") version "1.9.21"
-  id("name.remal.integration-tests") version "4.0.2"
-  id("jvm-test-suite")
```
Comment (andrewrlee) [inline build.gradle.kts:6]: don't think these plugins were needed

File: build.gradle.kts [hunk 5]
```diff
@@ -34,71 +23,45 @@ dependencies {
...
 }
-kotlin {
-  sourceSets["main"].apply {
-    kotlin.srcDir("${layout.buildDirectory}/generated/src/main/kotlin")
-  }
-}
-java {
-  sourceCompatibility = JavaVersion.VERSION_21
-  targetCompatibility = JavaVersion.VERSION_21
-}
```
Comment (andrewrlee) [inline build.gradle.kts:89]: not compiling any java

File: build.gradle.kts [hunk 6]
```diff
@@ -34,71 +23,45 @@ dependencies {
...
 
-repositories {
-  mavenCentral()
-}
```
Comment (andrewrlee) [inline build.gradle.kts:93]: this is applied by plugin

File: build.gradle.kts [hunk 7]
```diff
@@ -34,71 +23,45 @@ dependencies {
...
 tasks {
-  withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
-    kotlinOptions {
-      jvmTarget = "21"
-    }
-    withType<JavaCompile> {
-      sourceCompatibility = "21"
-    }
```
Comment (andrewrlee) [inline build.gradle.kts:101]: don't need to compile java

File: build.gradle.kts [hunk 8]
```diff
@@ -1,28 +1,17 @@
...
   testImplementation { exclude(group = "org.junit.vintage") }
 }
 
-testing {
-  suites {
-    val test by getting(JvmTestSuite::class) {
-      useJUnitJupiter()
-    }
-  }
-}
```
Comment (andrewrlee) [inline build.gradle.kts:24]: applied by hmpps gradle spring boot plugin

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationemploymentapi/config/ExceptionAdvice.kt [hunk 9]
```diff
@@ -112,19 +113,19 @@ class ControllerAdvice {
     log.info("Validation exception: ${e.message}", e)
     val genericMessage = "Unacceptable JSON " + e.message
     var errorDetails: String? = genericMessage
-    if (e.cause is MissingKotlinParameterException) {
-      val mkpx: MissingKotlinParameterException = e.cause as MissingKotlinParameterException
+    if (e.cause is MismatchedInputException) {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationemploymentapi/config/ExceptionAdvice.kt:116]: replacing deprecated class - `MissingKotlinParameterException` extends `MismatchedInputException` so this should still work

**Review by monskapinga — APPROVED** [review]
Have reviewed with @sacintha who's managed to build but unable to run due to environment issues.

**Pull request merged**